assets/js/theme/common/product-details.js

```diff
@@ -56,6 +56,27 @@ export default class ProductDetails {
         this.previewModal = modalFactory('#previewModal')[0];
     }
 
+    /**
+     * https://stackoverflow.com/questions/49672992/ajax-request-fails-when-sending-formdata-including-empty-file-input-in-safari
+     * Safari browser with jquery 3.3.1 has an issue uploading empty file parameters. This function removes any empty files from the form params
+     * @param form: FormData object
+     * @returns FormData object
+     */
+    filterEmptyFilesFromForm(form) {
+        try {
+            for (const [key, val] of form.entries()) {
+                if (val instanceof File) {
```

Comment:
might as well pull the second conditional into the first:
`if (val instanceof File && val.name === '' && val.size === 0) {`

could also simplify to `if (val instanceof File and !val.name && !val.size) {`